Config/CommonConfiguration.swift

```diff
@@ -82,6 +91,64 @@ class CommonConfiguration: NSObject, Configurable {
         MXKeyProvider.sharedInstance().delegate = EncryptionKeyManager.shared
     }
     
+    private func makeASCIIUserAgent() -> String? {
+        guard var userAgent = makeUserAgent() else {
+            return nil
+        }
+        if !userAgent.canBeConverted(to: .ascii) {
+            let mutableUserAgent = NSMutableString(string: userAgent)
+            if CFStringTransform(mutableUserAgent, nil, "Any-Latin; Latin-ASCII; [:^ASCII:] Remove" as CFString, false) {
+                userAgent = mutableUserAgent as String
+            }
+        }
+        return userAgent
+    }
+    
+    private func makeUserAgent() -> String? {
+        let clientName = Bundle.main.object(forInfoDictionaryKey: kCFBundleExecutableKey as String) as? String ?? Bundle.main.object(forInfoDictionaryKey: kCFBundleIdentifierKey as String) as? String ?? "unknown"
+        let clientVersion = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? Bundle.main.object(forInfoDictionaryKey: kCFBundleVersionKey as String) as? String ?? "unknown"
```

Comment:
I think `AppInfo.displayName` is possibly different from `Bundle.main.object(forInfoDictionaryKey: kCFBundleExecutableKey as String) as? String ?? Bundle.main.object(forInfoDictionaryKey: kCFBundleIdentifierKey as String) as? String ?? "unknown"` because it uses `CFBundleDisplayName`? I only changed it to read the version from AppInfo for now.

Reply:
Yes, it's different but may be more sensible. Executable name is constant (`Element`) even for forks unless they change it explicitly. It felt like we could use display name instead. I'll leave the decision to you.

Reply:
That's a good point actually. I changed it to read the name from `AppInfo`, too, now